### DISCUSSION TITLE
Do I actually need to unload textures?

### DISCUSSION THREAD

**Discussion by sakgoyal** [raysan5/raylib#4200] [General]
What happens if I omit `UnloadTexture()` calls in my C++ program that uses raylib?

I want to encapsulate the loadtexture behavior into a function but that will prevent me from being able to unload it at the end. 

But given I am using a modern OS (Win10), do I even need to unload it manually anymore? I would assume windows can handle memory cleanup at the end automatically right? 

current code:
```c++
void drawPlayer(const Texture2D& playerTexture) {
	drawtexturev(...);
	...
}

int main() {
...
const texture2d tex = loadtexture("...");
begindrawing();
drawPlayer(tex)
enddrawing();

unloadtexture(tex);
}
```

but I want to change it to this instead:
```c++
void drawPlayer() {
	static const texture2d tex = loadtexture("...");
	drawtexturev(...);
	...
}

int main() {
...
begindrawing();
drawPlayer()
enddrawing();

// no unloadtexture here
}
```

**Comment by ghost** [comment]


**Reply by sakgoyal** [reply]
1. but wouldnt the OS necessarily deallocate this at the end of the program? If I want the texture to exist for the entirety of the lifetime of the program, isnt the OS going to do this automatically?

**Reply by jamesl-github** [reply]
https://youtu.be/VdYOTsxgPu4?si=ksig87vOVZfSWD4E&t=544

**Reply by sakgoyal** [reply]
> [youtu.be/VdYOTsxgPu4?si=ksig87vOVZfSWD4E&t=544](https://youtu.be/VdYOTsxgPu4?si=ksig87vOVZfSWD4E&t=544)

this is exactly why I asked the question lmao. 
I watched the video and came straight here to ask.

**Comment by MrAdhit** [comment]
If the texture is used for the entire game lifetime, then no, your OS will deallocate it at the end of your game runtime.
If the texture is not used for the entire game lifetime, then yes, you need to deallocate it; otherwise, you'll have memory leaks.

**Reply by joaqu1m** [reply]
Nice answer!
